src/components/patch/PatchContainer.test.tsx

```diff
@@ -97,58 +96,42 @@ describe("PatchContainer", () => {
     expect(variant.state("sortedStatus")).toEqual([{ "count": 1, "status": "failed", "tasks": ["test-rest-client"] }, { "count": 5, "status": "success", "tasks": ["test-agent", "test-command", "test-rest-model", "test-rest-route", "test-auth"] }]);
   })
 
-  it("check that search returns correct results", () => {
-    const event = { currentTarget: { value: "change" } };
-    const expectedResults = ["5d700d89b237361b480b8488", "5d6ffa6897b1d37d85dae0bd", "5d6ff85e97b1d37d85dae045", "5d66d4aa97b1d3794298a82f"];
-    const notInResults = ["5d7148f497b1d36cde6c995e", "5d7018929dbe325975f6a049", "5d700fcb97b1d30d193967d0", 
-    "5d6fe06e9dbe3237b4cb7c15", "5d66f2e89dbe32552f1c91f5", "5d66f0fd9dbe32552f1c9166"];
-    const input = wrapper.find(".search-container").find(InputBase);
-    expect(input).toHaveLength(1);
-    input.prop("onChange")(event as React.ChangeEvent<HTMLInputElement>);
-    const visibleIds: string[] = [];
-    (wrapper.state("visiblePatches") as PatchInfo[]).map(patch => (
-      visibleIds.push(patch.id)
-    ));
-    for (const versionId of expectedResults) {
-      expect(visibleIds).toContain(versionId);
-    }
-    for (const versionId of notInResults) {
-      expect(visibleIds).not.toContain(versionId);
-    }
-    wrapper.setState({searchText: ""});
-  })
+  describe('Filtering pathces', () => {
+    let otherWrapper: enzyme.ReactWrapper
 
-  const checkFilteredState = jest.fn(() => {
-    expect(wrapper.state("selectedProjects")).toEqual([]);
-    expect(wrapper.state("selectedStatuses")).toEqual(["created"]);
-    const visibleIds: string[] = [];
-    const notInResults = ["5d7148f497b1d36cde6c995e", "5d700fcb97b1d30d193967d0", "5d700d89b237361b480b8488", 
-    "5d6ffa6897b1d37d85dae0bd", "5d6ff85e97b1d37d85dae045", "5d6fe06e9dbe3237b4cb7c15",
-    "5d66f2e89dbe32552f1c91f5", "5d66f0fd9dbe32552f1c9166", "5d66d4aa97b1d3794298a82f"];
-    const expectedResults = ["5d7018929dbe325975f6a049"];
-    (wrapper.state("visiblePatches") as PatchInfo[]).map(patch => {
-      visibleIds.push(patch.id);
-    });
-    for (const versionId of expectedResults) {
-      expect(visibleIds).toContain(versionId);
-    }
-    for (const versionId of notInResults) {
-      expect(visibleIds).not.toContain(versionId);
-    }
-  })
+    beforeEach(() => {
+      otherWrapper = enzyme.mount(<PatchContainer client={rest.EvergreenClient("", "", "", "", true)} username={"admin"} onFinishStateUpdate={null} />);
```

Comment:
It's not clear to me why we need two of these components. If you're worried about persistent state between tests, can we remount it between the test cases above?

Reply:
tldr: remounting the wrapper between test cases is one option. however these tests should be refactored and until they are, i think encapsulating the `beforeEach` and `afterEach` code to the filtering tests that require them is best. that way, the code required for these tests are clearly bound to the scope of the tests. 

ideally, at the top of the file, wrapper would be mounted and unmounted before and after each test using the jest `beforeEach` and `afterEach` to ensure that each test runs against a "clean" wrapper. the way the tests are currently written does not allow this to happen because the code [here](https://github.com/evergreen-ci/spruce/pull/25/files#diff-9adefdedf71f173747e15c1a2dae3b0cL15) is not encapsulated to a test scope and therefore throws an undefined error. tldr: these tests should be refactored.